### PR TITLE
[VEN-2565][VEN-2809]: Configure Prime and converters on arbitrum one

### PR DESCRIPTION
**Pull request by Debugger022**
## Description

Resolves VEN-2565 VEN-2809

### CODE REVIEW

File: multisig/proposals/arbitrumone/vip-018/commands.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  const config: CallPermission = {
+    target: ACM,
+    signature: "giveCallPermission(address,string,address)",
+    params: [target, signature, caller],
+  };
+
+  return config;
+};
+
+const incentiveAndAccessibility: IncentiveAndAccessibility = [0, 1];
```
Comment (chechu) [inline multisig/proposals/arbitrumone/vip-018/commands.ts:32]: Let's configure the same incentives already configured on Ethereum and BNB Chain: 1e14

File: multisig/simulations/arbitrumone/vip-018/index.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          }
+        });
+
+        it("Timelock should be the owner of ConverterNetwork", async () => {
+          expect(await converterNetwork.owner()).to.equal(arbitrumone.GUARDIAN);
+        });
+
+        it("Timelock should be the owner of XVSVaultTreasury", async () => {
+          expect(await xvsVaultTreasury.owner()).to.equal(arbitrumone.GUARDIAN);
+        });
```
Comment (chechu) [inline multisig/simulations/arbitrumone/vip-018/index.ts:87]: I would replace "Timelock" with "Guardian" in the descriptions, to be precise

File: simulations/vip-398/arbitrumone.ts [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+      }
+    });
+
+    it("Timelock should be the owner of ConverterNetwork", async () => {
+      expect(await converterNetwork.owner()).to.equal(arbitrumone.GUARDIAN);
+    });
+
+    it("Timelock should be the owner of XVSVaultTreasury", async () => {
+      expect(await xvsVaultTreasury.owner()).to.equal(arbitrumone.GUARDIAN);
+    });
```
Comment (chechu) [inline simulations/vip-398/arbitrumone.ts:149]: I would replace "Timelock" with "Guardian" in the descriptions, to be precise

File: simulations/vip-398/arbitrumone.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+      expect(await comptrollerLst.prime()).to.be.equal(ARBITRUM_PRIME);
+    });
+
+    it("Prime should contain correct markets", async () => {
+      expect((await prime.markets(ARBITRUM_VUSDC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VUSDT_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VWBTC_CORE))[4]).to.be.equal(true);
+      expect((await prime.markets(ARBITRUM_VWETH_LST))[4]).to.be.equal(true);
+    });
+
```
Comment (chechu) [inline simulations/vip-398/arbitrumone.ts:121]: We could add a check claiming a Prime token, impersonating one of the users who will be able to claim a Prime token as soon as the VIP is executed

File: multisig/proposals/arbitrumone/vip-018/addresses.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+// Function to filter assets based on a base asset
+const filterAssets = (assets: string[], baseAsset: string) => assets.filter(asset => asset !== baseAsset);
+
+export const USDTPrimeConverterTokenOuts = filterAssets(Assets, BaseAssets[0]);
+export const USDCPrimeConverterTokenOuts = filterAssets(Assets, BaseAssets[1]);
+export const WBTCPrimeConverterTokenOuts = filterAssets(Assets, BaseAssets[2]);
+export const WETHPrimeConverterTokenOuts = filterAssets(Assets, BaseAssets[3]);
+export const XVSVaultConverterTokenOuts = filterAssets(Assets, BaseAssets[4]);
+console.log("-------------- USDT", USDTPrimeConverterTokenOuts);
+console.log("-------------- USDC", USDCPrimeConverterTokenOuts);
```
[the remark this answers is not in the document]
Comment (Debugger022) [reply multisig/proposals/arbitrumone/vip-018/addresses.ts]: fixed, https://github.com/VenusProtocol/vips/pull/429/commits/563c701d661ec184a187dd111304ae60c8749e57

File: multisig/simulations/arbitrumone/vip-018/index.ts (hunk 2 above)
Comment (Debugger022) [reply multisig/simulations/arbitrumone/vip-018/index.ts]: fixed, https://github.com/VenusProtocol/vips/pull/429/commits/563c701d661ec184a187dd111304ae60c8749e57

File: simulations/vip-398/arbitrumone.ts (hunk 3 above)
Comment (Debugger022) [reply simulations/vip-398/arbitrumone.ts:149]: fixed, https://github.com/VenusProtocol/vips/pull/429/commits/563c701d661ec184a187dd111304ae60c8749e57

File: simulations/vip-398/arbitrumone.ts [hunk 6]
```diff
@@ -290,5 +298,27 @@ forking(276417316, async () => {
         parseUnits("1", 7),
       );
     });
+
+    it("claim prime token", async () => {
+      await network.provider.send("hardhat_setCode", [XVS_USER, "0x"]);
+      const xvsUserSigner = await initMainnetUser(XVS_USER, parseUnits("1"));
+
+      await expect(prime.connect(xvsUserSigner).claim()).to.be.revertedWithCustomError(prime, "Unauthorized");
+
```
Comment (GitGuru7) [inline simulations/vip-398/arbitrumone.ts:307]: Can we also verify after the user has deposited some XVS?

Comment (Debugger022) [reply simulations/vip-398/arbitrumone.ts]: done, https://github.com/VenusProtocol/vips/pull/429/commits/7c7baa1defabd4b994592d906d1d78ab88e73ea1

File: simulations/vip-398/arbitrumone.ts (hunk 4 above)
Comment (Debugger022) [reply simulations/vip-398/arbitrumone.ts]: done, https://github.com/VenusProtocol/vips/pull/429/commits/7c7baa1defabd4b994592d906d1d78ab88e73ea1

File: multisig/proposals/arbitrumone/vip-018/commands.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+  for (const converter of ConvertersArray) {
+    const config1 = grant(
+      converter,
+      "setConversionConfig(address,address,ConversionConfig)",
+      arbitrumone.NORMAL_TIMELOCK,
+    );
+    const config2 = grant(converter, "pauseConversion()", arbitrumone.NORMAL_TIMELOCK);
+    const config3 = grant(converter, "resumeConversion()", arbitrumone.NORMAL_TIMELOCK);
+    const config4 = grant(converter, "setMinAmountToConvert(uint256)", arbitrumone.NORMAL_TIMELOCK);
```
Comment (kkirka) [inline multisig/proposals/arbitrumone/vip-018/commands.ts:78]: I think now that we have omnichain governance in place we should configure all the permissions as specified in the docs (including the permissions to pause/unpause the conversions by fast-track/critical/guardian):

image

File: vips/vip-400/bscmainnet.ts [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+        signature: "giveCallPermission(address,string,address)",
+        params: [CONVERTER_NETWORK, "removeTokenConverter(address)", arbitrumone.NORMAL_TIMELOCK],
+        dstChainId: LzChainId.arbitrumone,
+      },
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [ARBITRUM_XVS_VAULT_TREASURY, "fundXVSVault(uint256)", arbitrumone.NORMAL_TIMELOCK],
+        dstChainId: LzChainId.arbitrumone,
+      },
```
Comment (kkirka) [inline vips/vip-400/bscmainnet.ts:129]: Should be available to fast track and critical timelocks as well

File: multisig/proposals/arbitrumone/vip-018/commands.ts (hunk 7 above)
Comment (chechu) [reply multisig/proposals/arbitrumone/vip-018/commands.ts:78]: In the VIP-245 and VIP-248 these permissions were set for the BNB Chain deployment. They could be used as a reference
* https://github.com/VenusProtocol/vips/blob/main/vips/vip-245/vip-245/vip-245.ts#L79
* https://github.com/VenusProtocol/vips/blob/main/vips/vip-248/vip-248/vip-248.ts#L129-L139

File: vips/vip-400/bscmainnet.ts (hunk 8 above)
Comment (chechu) [reply vips/vip-400/bscmainnet.ts]: https://github.com/VenusProtocol/vips/pull/429/commits/61a7f6fe6187c0dca945f029779a38cbc6589c00

File: multisig/proposals/arbitrumone/vip-018/addresses.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
+export const USDT = "0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9";
+export const USDC = "0xaf88d065e77c8cc2239327c5edb3a432268e5831";
+export const WBTC = "0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f";
+export const WETH = "0x82af49447d8a07e3bd95bd0d56f35241523fbab1";
```
[the remark this answers is not in the document]
Comment (chechu) [reply multisig/proposals/arbitrumone/vip-018/addresses.ts]: https://github.com/VenusProtocol/vips/pull/429/commits/2c3461f6b71d1b2b7bc90e13e4860cfe478fbc1a

File: vips/vip-400/bscmainnet.ts [hunk 10]
```diff
@@ -23,37 +20,108 @@ import {
...
 export const ARBITRUM_PLP = "0x86bf21dB200f29F21253080942Be8af61046Ec29";
 
-export const ARBITRUM_USDT = "0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9";
-export const ARBITRUM_USDC = "0xaf88d065e77c8cc2239327c5edb3a432268e5831";
-export const ARBITRUM_WBTC = "0x2f2a2543b76a4166549f7aab2e75bef0aefc5b0f";
-export const ARBITRUM_WETH = "0x82af49447d8a07e3bd95bd0d56f35241523fbab1";
+export const ARBITRUM_USDT = "0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9";
+export const ARBITRUM_USDC = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831";
+export const ARBITRUM_WBTC = "0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f";
+export const ARBITRUM_WETH = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1";
```
Comment (chechu) [inline vips/vip-400/bscmainnet.ts:31]: Using checksummed addresses, nothing else

Comment (Debugger022) [reply vips/vip-400/bscmainnet.ts]: done, https://github.com/VenusProtocol/vips/pull/429/commits/3c7166c46cbcf45ca2b0585fcef09e1b3b11fbc4

Comment (chechu) [reply vips/vip-400/bscmainnet.ts]: The addresses were already checksummed. We should revert https://github.com/VenusProtocol/vips/commit/3c7166c46cbcf45ca2b0585fcef09e1b3b11fbc4

Comment (Debugger022) [reply vips/vip-400/bscmainnet.ts:31]: done.

**Review by chechu — APPROVED** [review]

**Review by chechu — APPROVED** [review]

**Comment by chechu** [comment]
https://app.venus.io/#/governance/proposal/400?chainId=56

**Pull request merged**